File: Aman/Linked List/Remove K Digits.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
    string ans="";                                         
    for (int i=0;k>0 && i<num.size()-1;) 
    {
       if (num[i]>num[i+1]) 
       {
          num.erase(i,1);
          k--;         
          if (i>0) 
           i--; 
       }
       else
         i++;
   }

   if(k>0) 
       num.erase(num.size()-k,k);
        
   while (num.size()>0 && num[0]=='0')
      num.erase(0, 1);

   return num.size()>0?num:"0";
       
    }
};

File: Rohit Singhal/Stacks and Queues/RemoveKDigits.java
class Solution {
    public String removeKdigits(String num, int k) {
        
       Stack<Character> stack=new Stack<>();
        for(int i=0;i<num.length();i++)
        {
            char c=num.charAt(i);
            
             while(!stack.isEmpty() && stack.peek()>c && k>0)
            {
                stack.pop();
                k--;
            }
            if(!stack.isEmpty() || c!='0')
            stack.push(c);
            
            
        }
        while(!stack.isEmpty() && k!=0)
        {
            stack.pop();
            k--;
        }
        // String ans="";
        if(stack.isEmpty())
            return "0";
        
        StringBuilder sb = new StringBuilder();
        while(!stack.isEmpty())
            sb.append(stack.pop());
        sb.reverse();
        // while(!stack.isEmpty())
        // {
        //     ans=stack.pop()+ans;
        // }
       return sb.toString();
       
        
    }
}
